src/ASM/ASMs3Dmx.h

```diff
@@ -199,6 +199,14 @@ class ASMs3Dmx : public ASMs3D, private ASMmxBase
   //! \param[out] n3 Number of nodes in third (w) direction
   //! \param[in] basis Which basis to return size parameters for
   virtual bool getSize(int& n1, int& n2, int& n3, int basis = 0) const;
+protected:
+  //! \brief Returns the volume in the parameter space for an element.
+  //! \param[in] iel Element index
+  double getParametricVolume(int iel) const;
+  //! \brief Returns boundary face area in the parameter space for an element.
+  //! \param[in] iel Element index
+  //! \param[in] dir Local face index of the boundary face
+  double getParametricArea(int iel, int dir) const;
```

Comment:
And it is only called in specialized methods (integrate) so there is no need for the method to be virtual. But if you think it will ever be called in the parent context later..

-------- Opprinnelig melding --------
Fra: kmokstad <notifications@github.com>
Dato: 14.04.2016 17:43 (GMT+01:00)
Til: OPM/IFEM <IFEM@noreply.github.com>
Emne: Re: [OPM/IFEM] added: method to add a global lagrange multiplier to the system (#12)


In src/ASM/ASMs3Dmx.h<https://github.com/OPM/IFEM/pull/12#discussion_r59740756>:

> @@ -199,6 +199,14 @@ class ASMs3Dmx : public ASMs3D, private ASMmxBase
>    //! \param[out] n3 Number of nodes in third (w) direction
>    //! \param[in] basis Which basis to return size parameters for
>    virtual bool getSize(int& n1, int& n2, int& n3, int basis = 0) const;
> +protected:
> +  //! \brief Returns the volume in the parameter space for an element.
> +  //! \param[in] iel Element index
> +  double getParametricVolume(int iel) const;
> +  //! \brief Returns boundary face area in the parameter space for an element.
> +  //! \param[in] iel Element index
> +  //! \param[in] dir Local face index of the boundary face
> +  double getParametricArea(int iel, int dir) const;


This two methods overload the simular methods in ASMs3D, but then they need to be declared virtual both here and there, or else I'm nor sure you get the right one when invoked from a parent class.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/OPM/IFEM/pull/12/files/e4413e87fe4455ce9160c95fcc13c2455992a0c6..bff0ce9fa353fba985cba0d3c9cfa4ad48abed46#r59740756>